### PR TITLE
giph: Init at 1.1.1

### PR DESCRIPTION
**Pull request by legendofmiracles**
###### Motivation for this change
Adds giph to the repos

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Big thanks to オップナー2608 from the unofficial discord for writing the actual default.nix file, after I asked a nooby question :D

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117960` at 72597be7 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>giph</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '=(ffmpeg' doesn't match anything in file 'src/giph'
Near pkgs/applications/video/giph/default.nix:59:0:
```
   |
59 |     description = "Simple gif recorder";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/applications/video/giph/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  ]
+  ++ lib.optional
+    withSlop
+    slop
+  ++ lib.optional
+    withNotify
+    libnotify
+  ++ lib.optional
+    withPgrep
+    procps;
```
Comment (zseri) [inline pkgs/applications/video/giph/default.nix:49]: shell scripts don't need to declare buildInputs for the executables they use at runtime. see also: https://github.com/NixOS/nixpkgs/pull/117348 and  [Nix manual "complete dependencies"](https://nixos.org/manual/nix/stable/#idm140737322654464)

File: pkgs/applications/video/giph/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    rev = version;
+    sha256 = "19l46m1f32b3bagzrhaqsfnl5n3wbrmg3sdy6fdss4y1yf6nqayk";
+  };
+
+  postPatch = ''
+    substituteInPlace src/giph \
+      --replace '=(ffmpeg' '=(${ffmpeg}/bin/ffmpeg' \
+      --replace '="$(xdotool' '="$(${xdotool}/bin/xdotool' \
+  '' + lib.optionalString withSlop ''
+      --replace '=(ffmpeg' '=(${ffmpeg}/bin/ffmpeg' \
```
Comment (zseri) [inline pkgs/applications/video/giph/default.nix:30]: this line doesn't look correct, why does it replace ffmpeg when `withSlop` is checked?

File: pkgs/applications/video/giph/default.nix (hunk 1 above)
Comment (legendofmiracles) [reply pkgs/applications/video/giph/default.nix:49]: Baesd on your PR, I just shouldn't specify them at all in the buildInputs?
Only import them in the first line?

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -5384,6 +5384,12 @@
     githubId = 4158274;
     name = "Michiel Leenaars";
   };
+  legendofmiracles = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5387]: Please put this into a separate commit with the message ``maintainers: add legendofmiracles``.

File: pkgs/applications/video/giph/default.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  src = fetchFromGitHub {
+    owner = "phisch";
+    repo = "giph";
+    rev = version;
+    sha256 = "19l46m1f32b3bagzrhaqsfnl5n3wbrmg3sdy6fdss4y1yf6nqayk";
+  };
+
+  postPatch = ''
+    substituteInPlace src/giph \
```
Comment (SuperSandro2000) [inline pkgs/applications/video/giph/default.nix:26]: Wrapping the resulting binary is probably easier then replacing all hardcoded paths.

File: pkgs/applications/video/giph/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/applications/video/giph/default.nix:49]:
```suggestion
  ] ++ lib.optionals withSlop [ slop ]
  ++ lib.optionals withNotify [ libnotify ]
  ++ lib.optionals withPgrep [ procps ];
```

File: pkgs/applications/video/giph/default.nix [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+, xdotool
+, withSlop ? true
+, slop
+, withNotify ? true
+, libnotify
+, withPgrep ? true
+, procps
+}:
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/video/giph/default.nix:14]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

File: pkgs/applications/video/giph/default.nix [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{ stdenv
```
Comment (SuperSandro2000) [inline pkgs/applications/video/giph/default.nix:1]:
```suggestion
{ stdenvNoCC
```

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -2471,6 +2471,8 @@ in
 
   genromfs = callPackage ../tools/filesystems/genromfs { };
 
+  giph = callPackage ../applications/video/giph/default.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:2474]:
```suggestion
  giph = callPackage ../applications/video/giph/default.nix { };
```

File: pkgs/applications/video/giph/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/video/giph/default.nix:49]: They don't need to be at all in buildInputs. That is only required for autopatchelfHook

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
